lightwood/analysis/nc/icp.py

```diff
@@ -207,17 +209,18 @@ def predict(self, x, significance=None):
                 cal_scores = self.cal_scores[self.condition((x[j, :], c))][::-1]
                 n_cal = cal_scores.size
 
-                idx_left = np.searchsorted(cal_scores, nc, 'left')
-                idx_right = np.searchsorted(cal_scores, nc, 'right')
-                n_gt = n_cal - idx_right
-                n_eq = idx_right - idx_left + 1
-
-                p[j, i] = n_gt / (n_cal + 1)
+                n_eq = 0
+                n_gt = 0
+                for cal_score in cal_scores:
+                    if cal_score == nc:
+                        n_eq += 1
+                    elif nc < cal_score:
+                        n_gt += 1
 
                 if self.smoothing:
-                    p[j, i] += (n_eq * np.random.uniform(0, 1, 1)) / (n_cal + 1)
+                    p[j, i] = (n_gt + n_eq) / (n_cal + 1)
                 else:
-                    p[j, i] += n_eq / (n_cal + 1)
+                    p[j, i] = (n_gt + n_eq * random_uniform_dist) / (n_cal + 1)
```

Comment:
This is backwards, the smoothed version should have `n_eq * random_uniform_dist`, not the other way around.

Reply:
For the before/after. I printed the `labels` from the nc unittest.

This branch:

```
1st test:
[0.01005034 6.90775528 0.09622803 6.90775528]

2nd test:
[0.5        0.5        0.5        0.57781513 1.5        0.5       ]
```


Staging:

```
1st test:
[0.01005034 6.90775528 0.09622803 6.90775528]

2nd test:
[0.5        0.5        0.5        0.57781513 1.5        0.5       ]
```